src/spawn_players.c

```diff
@@ -531,100 +531,75 @@ void spawn_players_gp_one_player(f32* arg0, f32* arg1, f32 arg2) {
 
     D_8016556E = 0;
     if (gDemoMode == 1) {
-        spawn_player(gPlayerOneCopy, 0, arg0[D_80165270[0]], arg1[D_80165270[0]], arg2, 32768.0f,
+        spawn_player(gPlayerOne, 0, arg0[D_80165270[0]], arg1[D_80165270[0]], arg2, 32768.0f,
                      gCharacterSelections[0], PLAYER_HUMAN_AND_CPU);
-        spawn_player(gPlayerTwo, 1, arg0[D_80165270[1]], arg1[D_80165270[1]], arg2, 32768.0f, chooseCPUPlayers[0],
-                     PLAYER_EXISTS | PLAYER_CPU | PLAYER_START_SEQUENCE);
-        spawn_player(gPlayerThree, 2, arg0[D_80165270[2]], arg1[D_80165270[2]], arg2, 32768.0f, chooseCPUPlayers[1],
-                     PLAYER_EXISTS | PLAYER_CPU | PLAYER_START_SEQUENCE);
-        spawn_player(gPlayerFour, 3, arg0[D_80165270[3]], arg1[D_80165270[3]], arg2, 32768.0f, chooseCPUPlayers[2],
-                     PLAYER_EXISTS | PLAYER_CPU | PLAYER_START_SEQUENCE);
-        spawn_player(gPlayerFive, 4, arg0[D_80165270[4]], arg1[D_80165270[4]], arg2, 32768.0f, chooseCPUPlayers[3],
-                     PLAYER_EXISTS | PLAYER_CPU | PLAYER_START_SEQUENCE);
-        spawn_player(gPlayerSix, 5, arg0[D_80165270[5]], arg1[D_80165270[5]], arg2, 32768.0f, chooseCPUPlayers[4],
-                     PLAYER_EXISTS | PLAYER_CPU | PLAYER_START_SEQUENCE);
-        spawn_player(gPlayerSeven, 6, arg0[D_80165270[6]], arg1[D_80165270[6]], arg2, 32768.0f, chooseCPUPlayers[5],
-                     PLAYER_EXISTS | PLAYER_CPU | PLAYER_START_SEQUENCE);
-        spawn_player(gPlayerEight, 7, arg0[D_80165270[7]], arg1[D_80165270[7]], arg2, 32768.0f, chooseCPUPlayers[6],
-                     PLAYER_EXISTS | PLAYER_CPU | PLAYER_START_SEQUENCE);
+        for (int i = 1; i < NUM_PLAYERS; i++) {
+            spawn_player(&gPlayers[i], i, arg0[D_80165270[i]], arg1[D_80165270[i]], arg2, 32768.0f, chooseCPUPlayers[i - 1],
+                        PLAYER_EXISTS | PLAYER_CPU | PLAYER_START_SEQUENCE);
+        }
```

Comment:
Not totally sure I'm a fan of this change.

It's really easy to mess this part of the code up. Because I've done it before.